message-index/messages/GHC-64725/display/after/SomeDisplayLibrary.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module SomeDisplayLibrary where
+
+import GHC.TypeLits
+
+class Display a where
+  display :: a -> String
+
+instance Display () where
+  display () = "()"
+
+instance Display Bool where
+  display b = if b then "True" else "False"
+
+instance Display Int where
+  display = show
+
+instance (Display a, Display b) => Display (a, b) where
+  display (a, b) = "(" <> display a <> ", " <> display b <> ")"
+
+instance TypeError (Text "Cannot Display functions." :$$:
```

Comment:
```suggestion
instance TypeError (Text "Cannot display functions." :$$:
```

Reply:
ah, I went with capital D `Display` to allude to the type class, like the example from the [`GHC.TypeLits.TypeError` docs](https://hackage.haskell.org/package/base-4.16.1.0/docs/GHC-TypeLits.html#t:TypeError), but I don't have a strong preference